src/game/gamepadui/gamepadui_basepanel.cpp

```diff
@@ -59,6 +59,12 @@ GamepadUIFrame *GamepadUIBasePanel::GetCurrentFrame() const
 
 void GamepadUIBasePanel::SetCurrentFrame( GamepadUIFrame *pFrame )
 {
+    if (pFrame != NULL && m_pCurrentFrame != NULL)
```

Comment:
Are there possibly any cases where the setter is called with the same frame multiple times? I could see that causing the frame to close. Maybe the condition should also guarantee pFrame does not equal m_pCurrentFrame?

Reply:
I understand the concern, but `SetCurrentFrame` is only used when a frame is created. There is no possibility of it being called multiple times

Reply:
If you're sure! I think the name is a little misleading then. I would expect any method starting with "Set" to be a setter, which might be called at any time. But that's really neither here nor there